ProcessMaker/processmaker
# Fix Issues with Asset-Project Assignment and Project Selection

## Issue & Reproduction Steps
This PR addresses various issues encountered when assigning assets to projects via the asset creation modal and the configuration page. It also resolves complications related to populating the Project Select list with pre-configured projects on the asset configuration page.

## Solution

- Enhanced the relationships between the Project and Project Model on all asset models
- Refactored the synchronization of project assets on all asset models
- Improved handling for loading projects in the Project Select list

## How to Test

1. Ensure the existence of created projects
2. Create a new asset (e.g., processes, screens, scripts, data sources, decision tables)
3. Assign multiple projects to the newly created asset
4. Verify the presence of the asset in all selected projects
5. Navigate to the Asset designer page and select 'edit' or 'configure'
6. Ensure that the Project select list accurately displays all associated projects for the asset
7. Verify that adding or removing projects from the asset configuration page updates the asset accordingly in the selected projects.

## Related Tickets & Packages
- [FOUR-10981](https://processmaker.atlassian.net/browse/FOUR-10981)
- Projects Package: https://github.com/ProcessMaker/package-projects/pull/47
- Data Sources Package: https://github.com/ProcessMaker/package-data-sources/pull/332
- Decision Engine Package: https://github.com/ProcessMaker/package-decision-engine/pull/73

ci:next
ci:package-projects:observation/FOUR-10981
ci:package-data-sources:observation/FOUR-10981
ci:package-decision-engine:observation/FOUR-10981

## Code Review Checklist


[FOUR-10981]: https://processmaker.atlassian.net/browse/FOUR-10981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
